region/client.go

```diff
@@ -705,20 +700,24 @@ func (c *client) send(rpc hrpc.Call) (uint32, error) {
 	id := c.registerRPC(rpc)
 	header.CallId = &id
 
-	b := newBuffer(4)
-	defer func() { freeBuffer(b) }()
+	headerSize := proto.Size(header)
```

Comment:
It'd be nice to have comments (similar to the commit's comments) to explain some of the magic calculations going on here

Reply:
Will do.